### PR TITLE
V1.1 changes to Information endpoint

### PR DESCRIPTION
**Pull request by ashpatil-msft**
This PR contains the changes related to V1.1 schema updates for Information Endpoint.
- Added Unsupported and required fields introduced in V1.1 schema
- Added Source Agreements

Tests:
- Manually tested WinGet commands install, search, show and list with the endpoints and they are showing new data and installing packages as expected.

### CODE REVIEW

File: src/WinGet.RestSource/Models/Schemas/Information.cs [hunk 1]
```diff
@@ -20,8 +21,21 @@ public class Information
         /// </summary>
         public Information()
         {
-            this.SourceIdentifier = ApiConstants.SourceIdentifier;
+            this.SourceIdentifier = string.IsNullOrEmpty(ApiConstants.SourceIdentifier) ? "DefaultSourceIdentifier" : ApiConstants.SourceIdentifier;
```
Comment (jamespik) [inline src/WinGet.RestSource/Models/Schemas/Information.cs:24]: I assume you did this to make it easier to just test a source if it has not had all the proper environment variables set.  

I don't think this is the right way to go.  It bandaids over a configuration issue in a way that will cause distress if/when we have multiple sources all claiming to be the "DefaultSourceIdentifier".  Would we not prefer to detect this configuration error and inform the user of the correct way to configure the reference implementation?

File: src/WinGet.RestSource/Models/Schemas/Information.cs [hunk 2]
```diff
@@ -20,8 +21,21 @@ public class Information
         /// </summary>
         public Information()
         {
-            this.SourceIdentifier = ApiConstants.SourceIdentifier;
+            this.SourceIdentifier = string.IsNullOrEmpty(ApiConstants.SourceIdentifier) ? "DefaultSourceIdentifier" : ApiConstants.SourceIdentifier;
             this.ServerSupportedVersions = ApiConstants.ServerSupportedVersions;
+
+            // Example of how Unsupported Package match fields can be used.
+            this.UnsupportedPackageMatchFields = ApiConstants.UnSupportedPackageMatchFields;
+            this.RequiredPackageMatchFields = ApiConstants.RequiredPackageMatchFields;
```
Comment (jamespik) [inline src/WinGet.RestSource/Models/Schemas/Information.cs:29]: Its unclear that we would add unsupported fields to the stock reference implementation. The documentation Roy has been putting together is working towards how one would quickly and easily take the reference implementation and leverage it in their business.

Usage examples should be in documentation, not in the default implementation.

**Review by jamespik — CHANGES_REQUESTED** [review]

File: src/WinGet.RestSource/Models/Schemas/Information.cs (hunk 2 above)
Comment (ashpatil-msft) [reply src/WinGet.RestSource/Models/Schemas/Information.cs:33]: Removed the example

File: src/WinGet.RestSource/Models/Schemas/Information.cs (hunk 1 above)
Comment (ashpatil-msft) [reply src/WinGet.RestSource/Models/Schemas/Information.cs]: Fair point. I have added a check and exception to be thrown for this one.

File: src/WinGet.RestSource/Constants/ApiConstants.cs [hunk 3]
```diff
@@ -30,8 +30,33 @@ public class ApiConstants
         public static readonly ApiVersions ServerSupportedVersions = new ApiVersions()
         {
             "1.0.0",
+            "1.1.0",
         };
 
+        /// <summary>
+        /// Unsupported package match fields.
+        /// Note: NormalizedPackageNameAndPublisher field is added here to demonstrate how this property can be used.
```
[the remark this answers is not in the document]
Comment (ashpatil-msft) [reply src/WinGet.RestSource/Constants/ApiConstants.cs]: Currently Reference impl doesn't have NormalizedPackageNameAndPublisher field support. I have opened an issue for this for future extension.

File: src/WinGet.RestSource/Constants/ApiConstants.cs [hunk 4]
```diff
@@ -35,9 +35,9 @@ public class ApiConstants
 
         /// <summary>
         /// Unsupported package match fields.
-        /// Note: NormalizedPackageNameAndPublisher field is added here to demonstrate how this property can be used.
+        /// Note: NormalizedPackageNameAndPublisher field support is currently not implemented. There is a GitHub issue for this.
```
Comment (jamespik) [inline src/WinGet.RestSource/Constants/ApiConstants.cs:38]: Link or issue # might be appropriate here

File: src/WinGet.RestSource/Models/Schemas/Information.cs [hunk 5]
```diff
@@ -21,21 +21,18 @@ public class Information
         /// </summary>
         public Information()
         {
-            this.SourceIdentifier = string.IsNullOrEmpty(ApiConstants.SourceIdentifier) ? "DefaultSourceIdentifier" : ApiConstants.SourceIdentifier;
+            if (string.IsNullOrEmpty(ApiConstants.SourceIdentifier))
+            {
+                throw new System.ArgumentNullException("SourceIdentifier is not configured.");
```
Comment (jamespik) [inline src/WinGet.RestSource/Models/Schemas/Information.cs:26]: IIRC, the source identifier maps to the server identifier environment variable? Lets make the error message helpful in guiding folks to to the specific thing that needs to be updated/fixed to work with this setup.

File: src/WinGet.RestSource/Constants/ApiConstants.cs (hunk 4 above)
Comment (ashpatil-msft) [reply src/WinGet.RestSource/Constants/ApiConstants.cs]: Added the issue link

**Review by jamespik — APPROVED** [review]

**Pull request merged**